src/HttpSender.php

```diff
@@ -155,7 +155,10 @@ protected function createRequestOptions(PendingRequest $pendingRequest, RequestI
         // deal with PSR requests. We'll add the headers from the PSR
         // request to ensure the lowest-level values possible.
 
-        $config->add(RequestOptions::HEADERS, $psrRequest->getHeaders());
+        $config->add(RequestOptions::HEADERS, array_merge(
+            $psrRequest->getHeaders(),
+            $config->get(RequestOptions::HEADERS, [])
```

Comment:
It's hard to decide what should be treated as the main source of truth, should we be merging config headers into request headers and taking priority or should "headers" be priority? 🤔

Reply:
At first I thought the config headers would be more important. But after digging in, I see the *Connector headers* end up in `$config`, and the *Request headers* end up in `$psrRequest`. Also in `MergeRequestProperties`, the Request headers are merged into the Config headers, so also for consistency, I think it should be the other way around.